File: javascript/easy/palindromePermutation.js
/**
 * @param {string} s
 * @return {boolean}
 */
var canPermutePalindrome = function(s) {
    let map = new Map();
    for(let c of s) {
        if(map.get(c)) {
            map.delete(c);
        } else {
            map.set(c, true);
        }
    }
    return map.size <= 1;
};

File: ruby/easy/palindrome_permutation.rb
require 'set'
# @param {String} s
# @return {Boolean}
def can_permute_palindrome(s)
    set = Set.new
    s.each_char do |c|
        if set.include?(c)
            set.delete(c)
        else
            set.add(c)
        end
    end
    return set.length <= 1
end
